cns/service.go

```diff
@@ -63,19 +63,23 @@ func (service *Service) Initialize(config *common.ServiceConfig) error {
 		if err != nil {
 			return err
 		}
-
 		// Create the listener.
 		listener, err := acn.NewListener(u)
 		if err != nil {
 			return err
 		}
-
+		if config.TlsSettings.TlsEndpoint != "" {
+			// Start the listener and HTTP and HTTPS server.
+			if err = listener.StartTLS(config.ErrChan, config.TlsSettings); err != nil {
+				return err
+			}
+		}
 		// Start the listener.
-		err = listener.Start(config.ErrChan)
-		if err != nil {
+		// continue to listen on the normal endpoint for http traffic, this will be supported
+		// for sometime until partners migrate fully to https
```

Comment:
I think CNI could call CNS via HTTPS at one point, it will require though a slight change to the httpclient for CNI to include the subject name as part of the tlsconfig. But good point, CNI is a requirement here to remove this.